src/client/state.tsx

```diff
@@ -543,14 +551,23 @@ export function useServerStateRef<T>(
   useEffect(() => {
     const subscriber = (newState: SyncedState) => {
       const newSelectedState = selectorRef.current(newState);
-      if (!Object.is(newSelectedState, selectedStateRef.current)) {
+      if (
+        !(equalityFn ?? Object.is)(newSelectedState, selectedStateRef.current)
+      ) {
         selectedStateRef.current = newSelectedState;
         onChangeRef.current?.(newSelectedState);
       }
     };
     subscribe(subscriber);
     return () => unsubscribe(subscriber);
-  }, [onChangeRef, selectedStateRef, selectorRef, subscribe, unsubscribe]);
+  }, [
+    equalityFn,
```

Comment:
After re-reading my comment, I realize that it might not be super clear. Let me try again:
The first effect is supposed to subscribe to every server state change. On every change, it applies the selector and compares the selected part of state with the previously selected part of state using equalityFn. 
We basically never want to unsubscribe and resubscribe in this effect, which is why all dependencies (except for subscribe and unsubscribe themselves) should be refs.

However, what if equalityFn or the selector changes? Let's take this example:

```ts
function MyComponent({id}: {id: RRMapID}) {
  const map = useServerState(state => state.maps.entities[id]);

  // ... 
}
```

In this example, the selector is dependent on a prop. If we only had the first effect, the `map` would not update when the `id` passed to this component changes. That is what the second effect further down in `useServerStateRef` is for. It should run whenever `selector` or `equalityFn` changes (also see my other comment: https://github.com/cmfcmf/roc-and-roll/pull/137#discussion_r748877400) and checks if the selected part of the state is still equal to the previously selected part. If not, it executes the onChange callback. This is why we want the second effect to re-run whenever `selector` or `equalityFn` changes.

Reply:
Changes are now in. Would be great if you could take another look! The levels of indirection certainly contributed to some confusion :D

Reply:
Looks good now except for a minor issue (see other comment). Great work!